evap/results/tests/test_views.py

```diff
@@ -251,7 +251,57 @@ def test_private_course(self):
 class TestResultsTextanswerVisibility(WebTest):
     fixtures = ['minimal_test_data_results']
 
+    def test_textanswer_visibility_for_staff_before_publish(self):
+        page = self.app.get("/results/semester/1/course/1?public_view=false", user='staff')
+        self.assertIn(".course_orig_published.", page)
+        self.assertNotIn(".course_orig_hidden.", page)
+        self.assertNotIn(".course_orig_published_changed.", page)
+        self.assertIn(".course_changed_published.", page)
+        self.assertIn(".responsible_orig_published.", page)
+        self.assertNotIn(".responsible_orig_hidden.", page)
+        self.assertNotIn(".responsible_orig_published_changed.", page)
+        self.assertIn(".responsible_changed_published.", page)
+        self.assertIn(".responsible_orig_private.", page)
+        self.assertNotIn(".responsible_orig_notreviewed.", page)
+        self.assertIn(".contributor_orig_published.", page)
+        self.assertIn(".contributor_orig_private.", page)
+        self.assertIn(".other_responsible_orig_published.", page)
+        self.assertNotIn(".other_responsible_orig_hidden.", page)
+        self.assertNotIn(".other_responsible_orig_published_changed.", page)
+        self.assertIn(".other_responsible_changed_published.", page)
+        self.assertIn(".other_responsible_orig_private.", page)
+        self.assertNotIn(".other_responsible_orig_notreviewed.", page)
+
+    def test_textanswer_visibility_for_staff(self):
+        course = Course.objects.get(id=1)
+        course.publish()
+        course.save()
```

Comment:
can you try putting the two staff tests into a separate test class, creating the staff user in `setUpTestData`, and doing the course publish in this class in `setUpTestData` as well?